packages/preset-create-react-app/src/index.ts

```diff
@@ -1,20 +1,31 @@
 import { join, relative, resolve } from 'path';
 import { Configuration } from 'webpack';
 import { logger } from '@storybook/node-logger';
+import PnpWebpackPlugin from 'pnp-webpack-plugin';
 import { mergePlugins } from './helpers/mergePlugins';
-import { getReactScriptsPath } from './helpers/getReactScriptsPath';
+import {
+  getReactScriptsPath,
+  getReactScriptsPathWithYarnPnp,
+} from './helpers/getReactScriptsPath';
 import { processCraConfig } from './helpers/processCraConfig';
 import { checkPresets } from './helpers/checkPresets';
 import { getModulePath } from './helpers/getModulePath';
 import { Options } from './options';
 
 const CWD = process.cwd();
-const REACT_SCRIPTS_PATH = getReactScriptsPath();
+// When operating under PnP environments, this value will be set to a number
+// indicating the version of the PnP standard, see: https://yarnpkg.com/advanced/pnpapi#processversionspnp
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const IS_USING_YARN_PNP = (process.versions as any).pnp;
```

Comment:
We try very hard to avoid `any` in this codebase - can you try to define the type?

Reply:
This should also be converted to a boolean, as you aren't using the version.

Reply:
I added a commit to improve this by extending NodeJS type, with that no more `any`. I also improved the check used for `IS_USING_YARN_PNP `